core/include/gnuradio-4.0/Block.hpp

```diff
@@ -890,15 +909,20 @@ struct Block : protected std::tuple<Arguments...> {
 
             std::size_t in_min_samples = static_cast<std::size_t>(static_cast<double>(out_min_limit) / ratio);
             if (in_min_samples % denominator != 0) in_min_samples += denominator;
-            std::size_t in_min_wo_reminder = (in_min_samples / denominator) * denominator;
+            const std::size_t in_min_wo_remainder = (in_min_samples / denominator) * denominator;
 
-            const std::size_t in_max_samples     = static_cast<std::size_t>(static_cast<double>(out_max_limit) / ratio);
-            std::size_t       in_max_wo_reminder = (in_max_samples / denominator) * denominator;
+            const std::size_t in_max_samples      = static_cast<std::size_t>(static_cast<double>(out_max_limit) / ratio);
+            const std::size_t in_max_wo_remainder = (in_max_samples / denominator) * denominator;
 
-            if (ports_status.in_samples < in_min_wo_reminder) {
+            if (ports_status.in_samples < in_min_wo_remainder) {
                 return work::Status::INSUFFICIENT_INPUT_ITEMS;
             }
-            ports_status.in_samples  = std::clamp(ports_status.in_samples, in_min_wo_reminder, in_max_wo_reminder);
+
+            if (in_min_wo_remainder < in_max_wo_remainder) {
+                ports_status.in_samples = std::clamp(ports_status.in_samples, in_min_wo_remainder, in_max_wo_remainder);
+            } else {
+                ports_status.in_samples = std::min(ports_status.in_samples, std::min(in_min_wo_remainder, in_max_wo_remainder));
+            }
```

Comment:
I looked again to `std::clamp` docs and according to https://en.cppreference.com/w/cpp/algorithm/clamp: The behavior is undefined if the value of `lo` is greater than `hi`. Please check if my suggestion makes sense. With this change all tests passed.

```suggestion
            if (in_min_wo_remainder <= in_max_wo_remainder) {
                ports_status.in_samples = std::clamp(ports_status.in_samples, in_min_wo_remainder, in_max_wo_remainder);
            } else {
                return work::Status::ERROR;
            }
```